### PR TITLE
IDE-2144 Add new OSGI Module Wizard second page

### PR DESCRIPTION
**Pull request by simonjhy**

### CODE REVIEW

**Comment by gamerson** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: gradle/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/GradleProjectProvider.java [hunk 1]
```diff
@@ -82,52 +95,181 @@ public synchronized ILiferayProject provide( Object adaptable )
...
+
+                if( location != null && location.segmentCount() > 0 )
+                {
+                    if( !lastSegment.equals( projectName ) )
+                    {
+                        projecLocation = location.append( projectName );
+                    }
+                }
+
+                ProjectImportConfiguration configuration = new ProjectImportConfiguration();
```
Comment (gamerson) [inline gradle/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/GradleProjectProvider.java]: Lets find the java file to replace the properties first, and then import the gradle project.

File: gradle/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/GradleProjectProvider.java [hunk 2]
```diff
@@ -82,52 +95,181 @@ public synchronized ILiferayProject provide( Object adaptable )
...
+                        if( event.getResult().isOK() )
+                        {
+                            IProject moduleProject = CoreUtil.getProject( projectName );
+
+                            if( moduleProject.exists() )
+                            {
+                                List<IFile> files =
+                                    new SearchFilesVisitor().searchFiles( moduleProject, finalClassName );
+
+                                for( IFile file : files )
```
Comment (gamerson) [inline gradle/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/GradleProjectProvider.java]: Lets do this before the project import.

File: tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/modules/ComponentNameValidationService.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+import org.eclipse.sapphire.modeling.Status;
+import org.eclipse.sapphire.services.ValidationService;
+
+
+
+/** 
+ * @author Simon Jiang
+ */
+@SuppressWarnings( "restriction" )
+public class ComponentNameValidationService extends ValidationService
```
Comment (gamerson) [inline tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/modules/ComponentNameValidationService.java]: Lets add a ComponentNameDefaultValueService that makes the component name be something like this:
<Projectname>Portlet.java

Comment (gamerson) [inline tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/modules/ComponentNameValidationService.java]: Also lets add a PackageNameDefaultvalue Serivce that uses "com.example"

**Comment by gamerson** [comment]
Hey @simonjhy  we are getting closer, one or two more changes and this should be ready to go

**Comment by gamerson** [comment]
cc @jtydhr88

File: tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/modules/ComponentNameValidationService.java (hunk 3 above)
Comment (simonjhy) [inline tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/modules/ComponentNameValidationService.java]: you mean the component name should not diectly use project name as its default component name?

**Pull request closed without merging**